homeassistant/components/motion_blinds/__init__.py

```diff
@@ -62,21 +64,29 @@ def stop_motion_multicast(event):
 
     def update_gateway():
         """Call all updates using one async_add_executor_job."""
+        all_available = True
         motion_gateway.Update()
         for blind in motion_gateway.device_list.values():
             try:
                 blind.Update()
             except timeout:
                 # let the error be logged and handled by the motionblinds library
-                pass
+                all_available = False
 
-    async def async_update_data():
+        return all_available
+
+    async def async_update_data(self):
```

Comment:
Nothing is passed to the update_method from the coordinator, so this will not work as you intend it to do.

Reply:
Thanks for clarifying, I made some changes and now it schould work.
Could you revieuw again?

Reply:
There is nothing being passed to `self`

Reply:
now there is, I made a subclass of `DataUpdateCoordinator` -> `DataUpdateCoordinatorMotionBlinds`.
That overwrites the `_async_update_data` function to make sure it does pass `self`.

Reply:
could you have another look?

Reply:
Move `async_update_data` so that it is now a method on `DataUpdateCoordinatorMotionBlinds`, rather than passing in a `self` to a method that is not part of the class. (This technically works, but is not an expected style, so should be avoided).

Reply:
I don't think that is a good idea, async_update_data relays on `update_gateway` which in turn relays on `motion_gateway` which is only defined within the `async_setup_entry` function.
So I would also need to move the `DataUpdateCoordinatorMotionBlinds` class within the `async_setup_entry` function.
I think that might be possible, but I don't find it verry alligant.

Besides in this way I change the DataUpdateCoordinator class as little as possible (only adding in the self argument in order to be able to changes variables on the DataUpdateCoordinator class). Therefore I think the current method is more elegant.

Reply:
@MartinHjelmare the current code works, but I am not sure what would be the approprate style/way to achieve the same in the most ellegant way (changing `self.update_interval` of the `DataUpdateCoordinator` class within the update function).

Could you shed some light on how the preffered way of doing this would be?

Reply:
Pass the gateway to your custom coordinator class when instantiating that. Save the gateway as an instance attribute in the coordinator. Move the gateway update functions as methods in the coordinator.

Reply:
Thanks for the clarification @MartinHjelmare, I just made the changes.
Could you look if this is what you intended?